File: go/workflow.go
package demo

import (
	"time"

	"go.temporal.io/sdk/temporal"
	"go.temporal.io/sdk/workflow"
)

func Greeter(ctx workflow.Context, name string) error {
	ao := workflow.ActivityOptions{
		StartToCloseTimeout: 1 * time.Second,
		RetryPolicy: &temporal.RetryPolicy{
			InitialInterval: 1 * time.Second,
			MaximumInterval: 5 * time.Second,
		},
	}
	ctx = workflow.WithActivityOptions(ctx, ao)

	logger := workflow.GetLogger(ctx)

	logger.Info("Workflow started")

	var greeting string

	pick := workflow.ExecuteActivity(ctx,
		act.PickGreeting,
		name,
	)
	err := pick.Get(ctx, &greeting)
	if err != nil {
		return err
	}

	sms := workflow.ExecuteActivity(ctx,
		act.SendSMS,
		greeting,
		name,
	)

	email := workflow.ExecuteActivity(ctx,
		act.SendEmail,
		greeting,
		name,
	)

	err = sms.Get(ctx, nil)
	if err != nil {
		return err
	}

	err = email.Get(ctx, nil)
	if err != nil {
		return err
	}

	logger.Info("Workflow completed.")

	return nil
}

File: python/workflow.py
from datetime import timedelta
import asyncio

from temporalio import workflow
from activities import *
from temporalio.common import RetryPolicy

@workflow.defn
class Greeter:
    @workflow.run
    async def run(self, name: str) -> None:
        workflow.logger.info("Workflow started")

        greeting = await workflow.execute_activity(
            pickGreeting,
            PickGreetingInput(name),
            start_to_close_timeout=timedelta(seconds=1),
            retry_policy=RetryPolicy(maximum_interval=timedelta(seconds=5)),
        )

        sms = workflow.execute_activity(
            sendSMS,
            SendSMSInput(greeting, name),
            start_to_close_timeout=timedelta(seconds=1),
            retry_policy=RetryPolicy(maximum_interval=timedelta(seconds=5)),
        )

        email = workflow.execute_activity(
            sendEmail,
            SendEmailInput(greeting, name),
            start_to_close_timeout=timedelta(seconds=1),
            retry_policy=RetryPolicy(maximum_interval=timedelta(seconds=5)),
        )

        await asyncio.gather(sms, email)

        workflow.logger.info("Workflow completed")
